onnxruntime/core/providers/cpu/ml/tree_ensemble_common.h

```diff
@@ -123,6 +123,28 @@ Status TreeEnsembleCommon<InputType, ThresholdType, OutputType>::Init(const OpKe
       info.GetAttrsOrDefault<int64_t>("target_treeids"),
       info.GetAttrsOrDefault<float>("target_weights"),
       target_weights_as_tensor);
+#if !defined(ORT_MINIMAL_BUILD) || defined(ORT_EXTENDED_MINIMAL_BUILD)
+  if (status.IsOK()) {
+    std::vector<std::string> names = {"base_values", "nodes_falsenodeids", "nodes_featureids", "nodes_hitrates",
+                                      "nodes_missing_value_tracks_true", "nodes_modes", "nodes_nodeids", "nodes_treeids",
+                                      "nodes_truenodeids", "nodes_values",
+                                      "target_ids", "target_treeids", "target_nodeids", "target_weights"};
+    // #if !defined(ORT_MINIMAL_BUILD)
+    names.push_back("base_values_as_tensor");
+    names.push_back("nodes_hitrates_as_tensor");
+    names.push_back("nodes_values_as_tensor");
+    names.push_back("class_weights_as_tensor");
+    // #endif
+    for (auto name : names) {
```

Comment:
> for (auto name : names) { [](http://example.com/codeflow?start=3&length=26)

Suggestion: create a small function possibly templated that when the attribute name is queried, it can be registered as removable.

Reply:
The design has changed.